### PR TITLE
Add workaround for handling of unary minus operator in combination wi…

### PR DESCRIPTION
**Pull request by bjorncs**
…th `Long.MIN_VALUE`

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by bjorncs** [comment]
This is a workaround for usage of smallest signed 64-bit integer in grouping expressions (#26723) The grouping parser cannot handle negative numbers in expressions - they are represented as unsigned integers wrapped with `neg()`. An unary `-` operator can be forced by wrapping the operator and the single operand with parentheses. Otherwise it might be interpreted as a binary `-`.

File: container-search/src/main/javacc/com/yahoo/search/grouping/request/parser/GroupingParser.jj [hunk 1]
```diff
@@ -35,8 +35,14 @@ public class GroupingParser {
         if (str.indexOf('.') >= 0) {
             return Double.valueOf(str);
         }
+        if (str.equals(Long.toString(Long.MIN_VALUE).substring(1))) return Double.NaN;
         return Long.valueOf(str);
     }
+
+    private static GroupingExpression resolveNegation(GroupingExpression exp) {
+        if (exp instanceof DoubleValue && ((DoubleValue)exp).getValue().isNaN()) return new LongValue(Long.MIN_VALUE);
```
Comment (jonmv) [inline container-search/src/main/javacc/com/yahoo/search/grouping/request/parser/GroupingParser.jj:43]: `-NaN`? 🙈

Comment (jonmv) [reply container-search/src/main/javacc/com/yahoo/search/grouping/request/parser/GroupingParser.jj:43]: Hehe I guess we don't parse `NaN` ... looking at the above. Ok, this is the dirtiest PR I'll ever approve 🤣

**Review by jonmv — APPROVED** [review]
Cringe

File: container-search/src/main/javacc/com/yahoo/search/grouping/request/parser/GroupingParser.jj (hunk 1 above)
Comment (bjorncs) [reply container-search/src/main/javacc/com/yahoo/search/grouping/request/parser/GroupingParser.jj:43]: My original efforts all stranded with left-recursion which JavaCC does not support. So it ended up with this little hack sadly.

**Review by baldersheim — APPROVED** [review]

**Pull request merged**